### PR TITLE
storage_object_spec: updated to current rspec conventions

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

So this spec requires less cognitive load 

Part of #194.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (e.g. changes what is written to filesystem, or what is expected to be read from filesystem), run ***[integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** on stage as it tests preservation, and/or test in stage environment, in addition to specs.⚡

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -93,26 +93,20 @@ Naming/VariableNumber:
...
+# Offense count: 58
 # Configuration parameters: AssignmentOnly.
 RSpec/InstanceVariable:
   Exclude:
     - 'spec/unit_tests/moab/file_group_spec.rb'
-    - 'spec/unit_tests/moab/signature_catalog_spec.rb'
-    - 'spec/unit_tests/moab/storage_object_spec.rb'
-    - 'spec/unit_tests/moab/storage_object_version_spec.rb'
     - 'spec/unit_tests/moab/storage_repository_spec.rb'
     - 'spec/unit_tests/serializer/serializable_spec.rb'
```
Comment (ndushay) [inline .rubocop_todo.yml:111]: a form of progress -- these are some things that add to cognitive loading ...

File: spec/unit_tests/moab/storage_object_spec.rb [hunk 2]
```diff
@@ -1,551 +1,701 @@
...
+      expect(described_class.version_dirname(333)).to eq 'v0333'
+      expect(described_class.version_dirname(4444)).to eq 'v4444'
+      expect(described_class.version_dirname(55555)).to eq 'v55555'
+    end
   end
 
-  before(:all) do
-    @temp_dir_ingests = temp_dir.join('ingests')
-    @temp_object_dir = @temp_dir_ingests.join(BARE_TEST_DRUID)
-    @ingest_object_dir = ingests_dir.join(BARE_TEST_DRUID)
```
Comment (ndushay) [inline spec/unit_tests/moab/storage_object_spec.rb]: kill the instance vars!

File: spec/unit_tests/moab/storage_object_spec.rb [hunk 3]
```diff
@@ -1,551 +1,701 @@
...
-    @storage_object = described_class.new(FULL_TEST_DRUID, @temp_object_dir)
-    @storage_object.initialize_storage
-    @storage_object.storage_root = fixtures_dir.join('derivatives')
+    temp_object_dir.rmtree if temp_object_dir.exist?
   end
 
-  after(:all) do
-    @temp_dir_ingests.rmtree if !@temp_dir_ingests.nil? && @temp_dir_ingests.exist?
+  after do
+    temp_dir_ingests.rmtree if temp_dir_ingests&.exist?
```
Comment (justinlittman) [inline spec/unit_tests/moab/storage_object_spec.rb:21]: Wondering why `temp_dir_ingests` might be nil?

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comment.

File: spec/unit_tests/moab/storage_object_spec.rb (hunk 3 above)
Comment (ndushay) [reply spec/unit_tests/moab/storage_object_spec.rb:21]: valid question ...

**Pull request merged**